tutorials/epochs/plot_40_autogenerate_metadata.py

```diff
@@ -0,0 +1,449 @@
+"""
+.. _tut-autogenerate-metadata:
+
+Auto-generating ``Epochs`` metadata
+===================================
+
+This tutorial shows how to auto-generate metadata for `~mne.Epochs`, based on
+events via `mne.epochs.make_metadata`.
+
+We are going to use data from the :ref:`erp-core-dataset` (derived from
+:footcite:`Kappenman2021`). This is EEG data from a single participant
+performing an active visual task (Flankers task).
+
+.. note::
+   If you wish to skip the introductory parts of this tutorial, you may jump
+   straight to :ref:`tut-autogenerate-metadata-ern` after completing the data
+   import and event creation in the
+   :ref:`tut-autogenerate-metadata-preparation` section.
+
+This tutorial is loosely divided into two parts:
+
+1. We will first focus on producing ERP time-locked to the **visual
+   stimulation**, conditional on response correctness and response time in
+   order to familiarize ourselves with the `~mne.epochs.make_metadata`
+   function.
+2. After that, we will calculate ERPs time-locked to the **responses** – again,
+   conditional on response correctness – to visualize the error-related
+   negativity (ERN), i.e. the ERP component associated with incorrect
+   behavioral responses.
+
+.. contents:: Table of contents
+  :local:
+  :depth: 3
+
+.. _tut-autogenerate-metadata-preparation:
+
+Preparation
+^^^^^^^^^^^
+
+Data import and filtering
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's start by reading, filtering, and producing a simple visualization of the
+raw data. The data is pretty clean and contains very few blinks, so there's no
+need to apply sophisticated preprocessing and data cleaning procedures.
+"""
+
+from pathlib import Path
+import matplotlib.pyplot as plt
+import mne
+
+
+data_dir = Path(mne.datasets.erp_core.data_path())
```

Comment:
Is that an open question or are you asking me to do something? :) (I wouldn't know WHAT to do, exactly! But can have a look at the Circle config to hopefully figure things out)

Reply:
It's kind of an open question...  I thought we had something in place that pre-populated the circle system with all the datasets in advance, before any of the tutorials were run?  Or maybe that would be dumb because on most PR runs circle only builds 0-2 tutorials?

Reply:
> I thought we had something in place that pre-populated the circle system with all the datasets in advance

Just the ones that are used according to a suitable regex. You need to add `erp_core` to [`tools/circleci_download.sh`](https://github.com/mne-tools/mne-python/blob/main/tools/circleci_download.sh). For any new dataset my plan is to have people look at https://github.com/mne-tools/mne-python/pull/8867 to see the minimal set of files (9 including the tutorial that uses the dataset) that need to be changed to have a complete new-dataset PR, so it would be good to look to see if you've changed the appropriate set of files @hoechenberger

Reply:
Got it, will have a look.

Reply:
This is resolved now. Data should be downloaded by Circle before the tutorial is run.